Development

Ftmamhmod opened the pull request.

RawanSadek committed:
feat: implement routing and layout structure for the hotel system application

Ftmamhmod committed:
Refactor code structure for improved readability and maintainability

Ftmamhmod committed:
Implement code changes to enhance functionality and improve performance

RawanSadek committed:
feat: add logo image to assets for branding

RawanSadek committed:
feat: update authentication layout and styles

- Added @emotion/react and @emotion/styled dependencies for styling.
- Fixed typo in route path from "forget-password" to "forgot-password".
- Removed unused image files from the project.
- Added new background images for login, registration, and password recovery screens.
- Enhanced AuthLayout component to include dynamic background images and titles based on the current route.
- Introduced global CSS styles for consistent layout and background handling.

RawanSadek committed:
Merge pull request #2 from Ftmamhmod/RS/feat-AuthLayout

feat: update authentication layout and styles

RawanSadek committed:
feat: enhance login functionality and UI with MUI components, add loading indicator, and implement form validations

RawanSadek committed:
Merge pull request #3 from Ftmamhmod/NS/feat-login

feat: enhance login functionality and UI with MUI components, add loa…

RawanSadek committed:
feat: implement change password functionality with form validation and UI enhancements

RawanSadek committed:
feat: add change password functionality, update registration interface, and enhance validation rules

RawanSadek committed:
Merge pull request #4 from Ftmamhmod/RS/feat-Register

feat: add change password functionality, update registration interfac…

RawanSadek committed:
url updated

RawanSadek committed:
refactor: improve code formatting and add role input field in ChangePassword component

RawanSadek committed:
feat: update login and registration components for improved navigation and validation

RawanSadek committed:
Merge pull request #5 from Ftmamhmod/RS/feat-Masterlayout

feat: update login and registration components for improved navigatio…

RawanSadek committed:
feat: implement AuthContext for user authentication and session management

RawanSadek committed:
Merge pull request #6 from Ftmamhmod/RS/feat-AuthContect

feat: implement AuthContext for user authentication and session manag…

Ftmamhmod committed:
feat: enhance ForgotPassword and ResetPassword components with improved UI and validation

Ftmamhmod committed:
Implement code changes to enhance functionality and improve performance

Ftmamhmod committed:
Merge pull request #7 from Ftmamhmod/FM-feat/reset-and-forgot

Fm feat/reset and forgot

RawanSadek committed:
feat: update navigation logic in Login component based on user role

Ftmamhmod committed:
feat: update navigation logic in Login component and improve ResetPassword input IDs

Ftmamhmod committed:
feat: integrate i18next for multilingual support and add Arabic translations

Ftmamhmod committed:
Merge branch 'Development' into FM-feat/reset-and-forgot

Ftmamhmod committed:
Merge pull request #8 from Ftmamhmod/FM-feat/reset-and-forgot

Fm feat/reset and forgot

Ftmamhmod committed:
feat: add multilingual support for ChangePassword, ForgotPassword, and Login components with Arabic translations

RawanSadek committed:
feat: enhance Register component with profile image upload and role selection

RawanSadek committed:
Merge branch 'Development' into RS/feat-Register-modification

RawanSadek committed:
Merge pull request #10 from Ftmamhmod/RS/feat-Register-modification

feat: enhance Register component with profile image upload and role s…

RawanSadek committed:
feat: implement NotFound component with image display

RawanSadek committed:
Register page translated

Ftmamhmod committed:
feat: implement DeleteConfirmation component with dialog and translations

Ftmamhmod committed:
Merge pull request #11 from Ftmamhmod/FM/feat-delete-confirmation

feat: implement DeleteConfirmation component with dialog and translat…

mohamedsaeid1998 commented:
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/2365ac6f-2f0f-4254-9457-9644959cdb5f" />

Ftmamhmod committed:
hot fix

mohamedsaeid1998 commented:
<img width="307" height="91" alt="image" src="https://github.com/user-attachments/assets/07f70b9b-bdec-4c96-83c4-33c6b46358d2" />
All file extensions should always be lowercase (e.g., .tsx, .ts, .css, .json) and never capitalized. This avoids issues with case-sensitive file systems (like Linux servers) and keeps the project consistent and professional.

Ftmamhmod committed:
Rename END_POINTS.TS to END_POINTS.ts

Ftmamhmod committed:
Rename INTERFACES.TS to INTERFACES.ts

Ftmamhmod committed:
Rename VALIDATIONS.TS to VALIDATIONS.ts

mohamedsaeid1998 commented:
<img width="968" height="860" alt="image" src="https://github.com/user-attachments/assets/04318ca7-ff3e-49f6-a58c-d250a5af60f7" />
- The logo should be inside a link that redirects to the landing page.
- Placeholder text should be clearer, more visible, and descriptive (e.g., "Enter your email").
- Each input must have a linked label for better accessibility.
- Links to the register and forget password pages should have a distinct color to stand out.
- Add a link to return to the landing page.
- Reduce and align spacing between form elements for a cleaner layout.

RawanSadek commented:
> <img alt="image" width="968" height="860" src="https://private-user-images.githubusercontent.com/105775732/491946242-04318ca7-ff3e-49f6-a58c-d250a5af60f7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgzODI0ODksIm5iZiI6MTc1ODM4MjE4OSwicGF0aCI6Ii8xMDU3NzU3MzIvNDkxOTQ2MjQyLTA0MzE4Y2E3LWZmM2UtNDlmNi1hNThjLWQyNTBhNWFmNjBmNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyMFQxNTI5NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NmQxOTQ4YjgxYjg2ZjU2ZDhiMDhkZjQwYzRkYzkxZWM4MmFkZmMzYzk4OTM2ZDhjYzNlNGU5ZjNhM2JlNDY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.xQyLna4QXOFk-jUJlq0tySA7WMcK5_HIzGOy7ak-0ro"> - The logo should be inside a link that redirects to the landing page. - Placeholder text should be clearer, more visible, and descriptive (e.g., "Enter your email"). - Each input must have a linked label for better accessibility. - Links to the register and forget password pages should have a distinct color to stand out. - Add a link to return to the landing page. - Reduce and align spacing between form elements for a cleaner layout.

------------------
<img width="926" height="870" alt="image" src="https://github.com/user-attachments/assets/dc6ac6b5-b037-4875-b3e9-ae5f0b87cf46" />
Regarding the placeholders and Links, we are just following the figma design 👀. Should we apply these suggested changes even if they differ from the design, or should we stick to the figma?

mohamedsaeid1998 commented:
> > <img alt="image" width="968" height="860" src="https://private-user-images.githubusercontent.com/105775732/491946242-04318ca7-ff3e-49f6-a58c-d250a5af60f7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgzODI0ODksIm5iZiI6MTc1ODM4MjE4OSwicGF0aCI6Ii8xMDU3NzU3MzIvNDkxOTQ2MjQyLTA0MzE4Y2E3LWZmM2UtNDlmNi1hNThjLWQyNTBhNWFmNjBmNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyMFQxNTI5NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NmQxOTQ4YjgxYjg2ZjU2ZDhiMDhkZjQwYzRkYzkxZWM4MmFkZmMzYzk4OTM2ZDhjYzNlNGU5ZjNhM2JlNDY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.xQyLna4QXOFk-jUJlq0tySA7WMcK5_HIzGOy7ak-0ro"> - The logo should be inside a link that redirects to the landing page. - Placeholder text should be clearer, more visible, and descriptive (e.g., "Enter your email"). - Each input must have a linked label for better accessibility. - Links to the register and forget password pages should have a distinct color to stand out. - Add a link to return to the landing page. - Reduce and align spacing between form elements for a cleaner layout.
> 
> <img alt="image" width="926" height="870" src="https://private-user-images.githubusercontent.com/76558250/491948906-dc6ac6b5-b037-4875-b3e9-ae5f0b87cf46.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgzODM1NjAsIm5iZiI6MTc1ODM4MzI2MCwicGF0aCI6Ii83NjU1ODI1MC80OTE5NDg5MDYtZGM2YWM2YjUtYjAzNy00ODc1LWIzZTktYWU1ZjBiODdjZjQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTIwVDE1NDc0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhYWU1MWUzYWM5ODA5YTkyZWI3ODVhYjNjZjM2ZjkyODJmMDMxM2NmMDFlMWFhNjk4OTlkZTA1MzU5MzJmMGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.M3_AFoo2Z5kTSHNR8I0JTYz_4GYdjJ0xnlybOWG476o"> Regarding the placeholders and Links, we are just following the figma design 👀. Should we apply these suggested changes even if they differ from the design, or should we stick to the figma?

You don’t always have to strictly follow the design; sometimes the design itself has mistakes. It’s totally fine to improve and be creative with the UI — feel free to add enhancements and make it better than the original design.

mohamedsaeid1998 commented:
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/aeab3250-8c02-4fa3-babd-e6391bf2cff0" />
There’s no image showing on the register page

mohamedsaeid1998 commented:
<img width="597" height="119" alt="image" src="https://github.com/user-attachments/assets/16ef2f0a-98ab-4ab5-9cb4-4ef183f142c8" />
When selecting a value inside the select input, the validation error message does not disappear.

mohamedsaeid1998 commented:
<img width="611" height="180" alt="image" src="https://github.com/user-attachments/assets/f7c714ae-cfe5-4fd0-a99f-81e586fd68d1" />

The values inside the select options are written in capital letters, but when you select one, it appears in small letters.

mohamedsaeid1998 commented:
<img width="530" height="108" alt="image" src="https://github.com/user-attachments/assets/a64e2ab1-0e23-404c-b2a1-bb5a1653be6e" />

The displayed error message is incorrect

mohamedsaeid1998 commented:
<img width="532" height="175" alt="image" src="https://github.com/user-attachments/assets/95663e5a-ed0d-479d-94e7-50b9dd3fc707" />
You don’t need to apply the full backend password validation rules on the confirm password input. Its only purpose is to check if it matches the main password field, so just validate that the two values are identical.

mohamedsaeid1998 commented:
<img width="532" height="194" alt="image" src="https://github.com/user-attachments/assets/1da7b3cc-a562-47a1-a5b7-99257a308e80" />

mohamedsaeid1998 commented:
<img width="967" height="849" alt="image" src="https://github.com/user-attachments/assets/22fd03e0-5060-4ce9-895c-a2f593f49399" />
also in Reset Page

mohamedsaeid1998 commented on src/Modules/AuthModule/ChangePassword/ChangePassword.tsx:
```diff
@@ -1,8 +1,225 @@
+    formState: { errors, isSubmitting },
+  } = useForm<changePassDataTypes>();
+
+  const onSubmit = async (data: changePassDataTypes) => {
+    try {
+      const response = await axiosInstance.post(
+        USERS_URLS.CHANGE_PASSWORD,
+        data
+      );
```
A good practice when using the api response is to provide the correct type for it. This will make working with the response easier and less error prone.

mohamedsaeid1998 commented on src/Modules/AuthModule/ChangePassword/ChangePassword.tsx:
```diff
@@ -1,8 +1,225 @@
+      );
+      console.log(response);
+      toast.success(response.data.message);
+      navigate("/login");
```
- remove console.log
- We need to have optional chaining when accessing values coming from backend to avoid any runtime errors.

mohamedsaeid1998 commented on src/Modules/AuthModule/ChangePassword/ChangePassword.tsx:
```diff
@@ -1,8 +1,225 @@
+          )}
+        </FormControl>
+
+        <Button
+          type="submit"
+          disabled={isSubmitting}
+          variant="contained"
+          fullWidth
+          sx={{
+            bgcolor: "#3252DF",
+            marginTop: "60px",
+            marginBottom: "20px",
+            padding: "15px",
+            textTransform: "capitalize",
+            fontSize: "17px",
+          }}
+        >
```
it is not recommended to use pixels for font-sizes, you can read more about that here:
https://www.joshwcomeau.com/css/surprising-truth-about-pixels-and-accessibility

mohamedsaeid1998 commented on src/Modules/AuthModule/Login/Login.tsx:
```diff
@@ -1,8 +1,192 @@
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<loginDataTypes>();
+
+  const onSubmit = async (data: loginDataTypes) => {
+    try {
+      const response = await axiosInstance.post(USERS_URLS.LOGIN, data);
+      // console.log(response);
+      localStorage.setItem(
+        "token",
+        response.data.data.token.replace(/^Bearer\s+/, "")
+      );
+      localStorage.setItem("userData", JSON.stringify(response.data.data.user));
+      getLoginData();
+      toast.success(`Welcome to StayCation!`);
+      if (response.data.data.user.role === "admin") {
+        navigate("/dashboard");
+        return;
+      } else if (response.data.data.user.role === "user") {
+        navigate("/landing-page");
+      }
+      if (response.data.data.user.role == "admin") navigate("/dashboard");
+      else navigate("/landing-page");
+    } catch (err) {
+      const error = err as AxiosError<{ message: string }>;
+      toast.error(error.response?.data?.message || "Something went wrong");
+    }
```
why another if ???

mohamedsaeid1998 commented on src/Modules/AuthModule/ResetPassword/ResetPassword.tsx:
```diff
@@ -1,8 +1,251 @@
+        {/* Email */}
+        <FormControl fullWidth>
+          <Typography color="#152C5B">Email Address</Typography>
+          <FilledInput
+            {...register("email", EMAIL_VALIDATION)}
+            id="email"
+            placeholder="Please type here..."
+            value={watch("email")}
+            disabled
+            readOnly
+            disableUnderline
+            sx={{
+              bgcolor: "#F5F6F8",
+              borderRadius: "4px",
+              marginTop: "10px",
+              "& .MuiInputBase-input": {
+                py: "10px",
+                "::placeholder": {
+                  color: "#b4b4b4ff",
+                },
+              },
+            }}
+          />
+          {errors.email && (
+            <Typography sx={{ color: "red" }}>
+              {errors.email.message as string}
+            </Typography>
```
There’s no need to show an error message under the email input since it will always be pre-filled with the user’s email coming from the forget password page. Also, make sure the button has a cursor: not-allowed style when it’s disabled to give clear feedback to the user.

mohamedsaeid1998 commented on src/Modules/AuthModule/Register/Register.tsx:
```diff
@@ -1,8 +1,450 @@
+            {/* Phone number */}
+            <FormControl fullWidth>
+              <Typography color="#152C5B">{t('Phone_Number')}</Typography>
+              <FilledInput
+                {...register("phoneNumber", PHONE_VALIDATION)}
+                id="phoneNumber"
+                placeholder={t('Type_Here')}
+                disableUnderline
+                sx={{
+                  bgcolor: "#F5F6F8",
+                  borderRadius: "4px",
+                  marginTop: "10px",
+                  "& .MuiInputBase-input": {
+                    py: "10px",
+                    "::placeholder": {
+                      color: "#b4b4b4ff",
+                    },
+                  },
+                }}
+              />
+              {errors.phoneNumber && (
```
Phone number should be of type tel.

RawanSadek committed:
feat: add i18next-browser-languagedetector for language detection and update translations
feat: implement RoomsList component with loading state and pagination
fix: update ChangePassword component for validation and placeholder text
style: adjust CSS for multilingual support in auth background

Ftmamhmod committed:
Add interfaces for auth, rooms and components

Ftmamhmod committed:
Merge pull request #12 from Ftmamhmod/RS/feat-Rooms

feat: add i18next-browser-languagedetector for language detection and…

RawanSadek committed:
feat: update authentication context to use userData and enhance RoomsList with action menu and pagination

RawanSadek committed:
Merge pull request #13 from Ftmamhmod/RS/feat-RoomsList

feat: update authentication context to use userData and enhance Rooms…

RawanSadek committed:
feat: integrate Controller from react-hook-form for role selection and update translation keys for consistency

RawanSadek committed:
feat: enhance password confirmation validation and update role selection handling in ChangePassword component

NadaSoliman29 committed:
create sidebar , navbare , userfolder

Ftmamhmod committed:
feat: facilites crud

Ftmamhmod committed:
new

Ftmamhmod committed:
Merge branch 'Development' into FM/feat-facilites-crud

Ftmamhmod committed:
Merge pull request #14 from Ftmamhmod/FM/feat-facilites-crud

Fm/feat facilites crud

Ftmamhmod committed:
fix

RawanSadek committed:
booking data pop up done

RawanSadek committed:
Merge branch 'Development' into RS/feat-BookingList

RawanSadek committed:
Merge pull request #15 from Ftmamhmod/RS/feat-BookingList

booking data pop up done

RawanSadek committed:
added comma

Ftmamhmod committed:
feat: add ads table

Ftmamhmod committed:
new

Ftmamhmod committed:
Merge branch 'main' into FM/feat-ADS-crud

Ftmamhmod committed:
Merge pull request #17 from Ftmamhmod/FM/feat-ADS-crud

Fm/feat ads crud

Ftmamhmod committed:
test

Ftmamhmod committed:
Delete src/Services/INTERFACES.TS

Ftmamhmod committed:
fix

Ftmamhmod committed:
fix: delete

Ftmamhmod committed:
Merge pull request #18 from Ftmamhmod/FM/fix-delete-ads-and-facilities

fix: delete

Ftmamhmod committed:
Feat:dashboard

Ftmamhmod committed:
feat:userTable

Ftmamhmod committed:
Merge pull request #19 from Ftmamhmod/FM/fix-delete-ads-and-facilities

Fm/fix delete ads and facilities

NadaSoliman29 committed:
add userlist

NadaSoliman29 committed:
new

NadaSoliman29 committed:
Merge branch 'Development' into NS-feat-Users

NadaSoliman29 committed:
Merge pull request #20 from Ftmamhmod/NS-feat-Users

Ns feat users

Ftmamhmod committed:
fix: open popup

Ftmamhmod committed:
Merge branch 'Development' of https://github.com/Ftmamhmod/hotel-system into Development

RawanSadek committed:
rooms data done

RawanSadek committed:
done

RawanSadek committed:
Merge branch 'Development' into RS/feat-RoomsActions

RawanSadek committed:
Merge pull request #21 from Ftmamhmod/RS/feat-RoomsActions

rooms data done

RawanSadek committed:
updated

Ftmamhmod committed:
fix: facilities

RawanSadek committed:
hotfix

RawanSadek committed:
Merge branch 'Development' of https://github.com/Ftmamhmod/hotel-system into Development

RawanSadek committed:
updated

Ftmamhmod committed:
edit: dashboard colors

Ftmamhmod committed:
refactor: update state management and improve form handling in various components

Ftmamhmod committed:
fix : edit facility

Ftmamhmod committed:
fix: edit ads

RawanSadek committed:
updated

RawanSadek committed:
Merge branch 'Development' into RS/feat-Rooms-Updates

RawanSadek committed:
Merge pull request #22 from Ftmamhmod/RS/feat-Rooms-Updates

updated

Ftmamhmod committed:
fix: remove unused translation hook and update loading indicator

Ftmamhmod committed:
feat: enhance authentication context and user profile management; update footer and navbar components

Ftmamhmod committed:
Merge pull request #23 from Ftmamhmod/FM/feat-navbar-and-footer

feat: enhance authentication context and user profile management; upd…

Ftmamhmod committed:
fix: update favorite list route  '/favourits'

Ftmamhmod committed:
feat: add translation option to MasterNavbar menu

NadaSoliman29 committed:
edit some issues

NadaSoliman29 committed:
Merge branch 'Development' of https://github.com/Ftmamhmod/hotel-system into Development

NadaSoliman29 committed:
edit some issues

NadaSoliman29 committed:
added the roomdetails and datepicker library

Ftmamhmod committed:
feat: add Home component for landing page with booking functionality

Ftmamhmod committed:
Merge pull request #24 from Ftmamhmod/FM/feat-home-top-section

feat: add Home component for landing page with booking functionality

NadaSoliman29 committed:
add room-deatils and solve some issues

NadaSoliman29 committed:
.

NadaSoliman29 committed:
editname of room

Ftmamhmod committed:
feat: Add AdsSwiper component for displaying ads and rooms in a slider format

feat: Implement FavCard component for displaying favorite rooms with interaction options

feat: Create MostAds component to showcase the most popular ads with favorite functionality

feat: Develop ReviewCard component for displaying reviews with images and ratings

feat: Introduce RoomCard component for displaying room details with hover effects and favorite option

Ftmamhmod committed:
style: Update color scheme to use hex values instead of CSS variables

RawanSadek committed:
add/remove favs done

RawanSadek committed:
updated

NadaSoliman29 committed:
edit some features

NadaSoliman29 committed:
Merge branch 'Development' of https://github.com/Ftmamhmod/hotel-system into Development

Ftmamhmod committed:
fix: Adjust layout and dimensions in FavCard and MostAds components for better responsiveness

Ftmamhmod committed:
Merge branch 'Development' into FM/feat-home

Ftmamhmod committed:
Merge pull request #25 from Ftmamhmod/FM/feat-home

Fm/feat home

Ftmamhmod committed:
feat: Add AdminLayout component to dashboard route

Ftmamhmod committed:
feat: Update room fetching to use USER_ROOMS_URLS in AdsSwiper and MostAds components

RawanSadek committed:
updated

RawanSadek committed:
Merge branch 'Development' of https://github.com/Ftmamhmod/hotel-system into Development

RawanSadek committed:
handle adding to favourites in landing page

NadaSoliman29 committed:
added payment page

NadaSoliman29 committed:
Merge branch 'Development' of https://github.com/Ftmamhmod/hotel-system into Development

NadaSoliman29 committed:
edit payment isuues

RawanSadek committed:
payment updated

RawanSadek committed:
updated
